### PR TITLE
feat: implemented JWT support and Swagger UI configuration

### PR DESCRIPTION
**Pull request by PavloMelnyk10**
- Added JWT-based authentication support
- Created `AuthenticationService`, `JwtAuthenticationFilter`, and `JwtUtil` classes
- Updated `AuthenticationController` with `/api/auth/login` endpoint
- Added Swagger UI with JWT authentication via `OpenApiConfig` class

### CODE REVIEW

File: src/main/java/mate/academy/bookstore/config/OpenApiConfig.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class OpenApiConfig {
+
+    @Bean
+    public OpenAPI customOpenApi() {
+        return new OpenAPI()
+            .components(new Components()
+                    .addSecuritySchemes("BearerAuth",
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstore/config/OpenApiConfig.java]: create constants for strings

File: src/main/java/mate/academy/bookstore/controller/AuthController.java [hunk 2]
```diff
@@ -27,4 +31,12 @@ public UserResponseDto register(@Valid @RequestBody UserRegistrationRequestDto r
             throws RegistrationException {
         return userService.register(requestDto);
     }
+
+    @PostMapping("/login")
+    @Operation(summary = "Authenticate user",
+            description = "Authenticate user using email "
+                    + "and password and return JWT token for accessing secure endpoints")
+    public UserLoginResponseDto login(@RequestBody UserLoginRequestDto requestDto) {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstore/controller/AuthController.java]:
```suggestion
    public UserLoginResponseDto login(@RequestBody @Valid UserLoginRequestDto requestDto) {
```

File: src/main/java/mate/academy/bookstore/dto/user/UserLoginRequestDto.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+import jakarta.validation.constraints.Email;
+import jakarta.validation.constraints.NotBlank;
+import org.hibernate.validator.constraints.Length;
+
+public record UserLoginRequestDto(
+        @NotBlank
+        @Length(min = 8, max = 35)
+        @Email
+        String email,
+        String password
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstore/dto/user/UserLoginRequestDto.java:11]: validate password as well

File: src/main/java/mate/academy/bookstore/security/JwtAuthenticationFilter.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+            Authentication authentication = new UsernamePasswordAuthenticationToken(
+                    userDetails, null, userDetails.getAuthorities());
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+        }
+
+        filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader("Authorization");
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstore/security/JwtAuthenticationFilter.java]:
```suggestion
        String bearerToken = request.getHeader(HttpHeaders.AUTHORIZATION);
```

File: src/main/java/mate/academy/bookstore/security/JwtAuthenticationFilter.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith("Bearer ")) {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstore/security/JwtAuthenticationFilter.java]: create constant for `"Bearer "`

File: src/main/java/mate/academy/bookstore/security/JwtAuthenticationFilter.java [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+            return bearerToken.substring(7);
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstore/security/JwtAuthenticationFilter.java]:
```suggestion
            return bearerToken.substring(constant.length());
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/mate/academy/bookstore/dto/user/UserLoginRequestDto.java [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+import jakarta.validation.constraints.Email;
+import jakarta.validation.constraints.NotBlank;
+import org.hibernate.validator.constraints.Length;
+
+public record UserLoginRequestDto(
+        @NotBlank
+        @Length(min = 8, max = 35)
+        @Email
+        String email,
+        @NotBlank
```
Comment (liliia-ponomarenko) [inline src/main/java/mate/academy/bookstore/dto/user/UserLoginRequestDto.java:10]: Should you validate the length?

**Review by liliia-ponomarenko — APPROVED** [review]
Good job! 👍

File: src/main/java/mate/academy/bookstore/dto/user/UserLoginRequestDto.java (hunk 7 above)
Comment (PavloMelnyk10) [reply src/main/java/mate/academy/bookstore/dto/user/UserLoginRequestDto.java:10]: I think no, it's not necessary here. I've removed it.

**Pull request merged**